JsonPath/IPathFunctionDefinition.cs

```diff
@@ -33,6 +34,10 @@ private class NothingValue{}
 	/// <summary>
 	/// Represents the absence of a JSON value and is distinct from any JSON value, including null.
 	/// </summary>
+	[UnconditionalSuppressMessage("Trimming", "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+		Justification = "This type is not returned to the user and should never be serialized.")]
+	[UnconditionalSuppressMessage("AOT", "IL3050:Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.",
+		Justification = "This type is not returned to the user and should never be serialized.")]
 	public static JsonValue Nothing { get; } = JsonValue.Create(new NothingValue())!;
```

Comment:
There's an AOT-friendly way to do this, if you get a JsonTypeInfo of NothingValue then you can go through an overload of JsonValue.Create that doesn't trip the warnings.

Something like this would do it:
```C#
/// <summary>
/// Base class for defining an expression function which returns `ValueType`.
/// </summary>
public abstract partial class ValueFunctionDefinition : IReflectiveFunctionDefinition, IPathFunctionDefinition
{
	private class NothingValue{}

	[JsonSerializable(typeof(NothingValue))]
	private partial class NothingValueSerializerContext : JsonSerializerContext
	{

	}

	/// <summary>
	/// Represents the absence of a JSON value and is distinct from any JSON value, including null.
	/// </summary>
	public static JsonValue Nothing { get; } = JsonValue.Create(new NothingValue(), NothingValueSerializerContext.Default.NothingValue)!;

```